tests/test_process_labels_fusion.py

```
@@ -0,0 +1,539 @@
+        (_make_gemma3n, legacy_gemma3n_process_labels, [130, 131, 132, 133], "gemma3n"),
+        (_make_gemma4, legacy_gemma4_process_labels, [140, 141, 142, 143, 150, 151, 160], "gemma4"),
+        (_make_llama32v, legacy_base_process_labels, [], "llama32v"),
+        (_make_llama4, legacy_base_process_labels, [], "llama4"),
+        (_make_pixtral, legacy_base_process_labels, [], "pixtral"),
+        (_make_mistral_v7, legacy_base_process_labels, [], "mistral_v7"),
+    ],
+)
+@pytest.mark.parametrize("seed", list(range(50)))
+def test_process_labels_parity_fuzz(strategy_factory, legacy_fn, extra_ids, name, seed):
+    """50 random batches per strategy: legacy vs fused must produce byte-identical labels."""
+    rng = random.Random(seed * 7919 + hash(name) % 1000)
```

Ruff (0.15.13)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)